modmesh/pilot/_base_app.py

```diff
@@ -0,0 +1,869 @@
+
+        single_button = QPushButton("Single")
+        single_button.clicked.connect(self.app.single_layout)
+        hbox.addWidget(single_button)
+
+        layout.addLayout(hbox)
+
+        data_line_config_label = QLabel("Data line configuration")
+        layout.addWidget(data_line_config_label)
+
+        table = QTableView()
+        table_model = ConfigTableModel(self.app.plot_config)
+        table.setModel(table_model)
+        layout.addWidget(table)
+
+        save_button = QPushButton("Save")
+        save_button.clicked.connect(self.app.update_layout)
+        layout.addWidget(save_button)
+
+        self.setLayout(layout)
+
+
+class ConfigWindow(QWidget):
+    """
+    ConfigWindow class for managing solver and plotting area configurations.
+
+    This class inherit from the QWidget class and provides a graphical user
+    interface for managing solver configurations. It includes options to set
+    the solver type, view and edit configuration parameters, and control the
+    solver's behavior.
+
+    Attributes:
+        - `app`: The app want to plot something in plotting area.
+        - `solver_config` (:class:`SolverConfig`): Configuration object
+          for the solver.
+        - `plot_config` (:class:`PlotConfig`): Configuration object
+          for the plotting area.
+
+    Methods:
+        - :meth:`setup()`: Setup method to configure the window.
+        - :meth:`on_open()`: Callback function when plot configure modal
+          window is opened.
+        - :meth:`on_close()`: Callback function when plot configure modal
+          windows is closed.
+        - :meth:`init_ui()`: Define the GUI layout of the window.
+    """
+
+    def __init__(self, app, parent=None):
+        super().__init__(parent)
+        self.app = app
+        self.plot_config_open = False
+        self.setup()
+        self.init_ui()
+
+    def setup(self):
+        """
+        Assign the configure object from app
+
+        :return: nothing
+        """
+        self.solver_config = self.app.solver_config
+        self.plot_config = self.app.plot_config
+
+    def on_open(self):
+        self.plot_config_open = True
+        if not self.plot_config_dialog:
+            self.plot_config_dialog = PlotConfigDialog(self.app, self)
+        self.plot_config_dialog.exec_()
+
+    def on_close(self):
+        self.plot_config_open = False
+        if self.plot_config_dialog:
+            self.plot_config_dialog.close()
+
+    def init_ui(self):
+        """
+        Define the GUI layout of the window for this class
+
+        :return: QWidget
+        """
+        main_layout = QVBoxLayout(self)
+
+        # First VBox, handle operation about solver configuration, weight=4
+        vbox1 = QVBoxLayout()
+        vbox1.setStretch(0, 4)
+
+        solver_label = QLabel("Solver")
+        vbox1.addWidget(solver_label)
+
+        combo_box = QComboBox()
+        combo_box.addItem("Euler1D-CESE")
+        vbox1.addWidget(combo_box)
```

Comment:
The current implementation hardcodes "Euler1D-CESE" in the combo box. We should modify this to show the solver in use, or potentially support selecting between multiple solvers if needed.